atlp-rwanda/atlp-pulse-mobile
# #67 Setup in-app notifications

### What does this PR do?

This PR includes setup of in app notifications that can be triggered on backend.

### Description of Task to be completed?

If the backend is about to send email notifications it will also trigger a notification on the app. The app has filters to know which notification to show and when it is launched it starts listening to notification that might target it.

### How should this be manually tested?

- Open the app and go to [this site](https://expo.dev/notifications) to trigger a notification
- Use this push token to trigger a notification: `ExponentPushToken[voKvU_GA9HRgVVF9RP14px]`
- For data (JSON) use this: `{"criteria":{"type":"PUBLIC"}}`
- Other fields are intuitive you can feel them with what you want.
- After that click on `Send Notification` and check on your device if the notification comes.

### Screenshots

<img width="300" src="https://github.com/user-attachments/assets/d6bacc0c-7620-4f2c-89aa-737d7a500ddf" />
